lib/core/src/Cardano/Wallet.hs

```diff
@@ -751,6 +756,32 @@ estimateTxFee ctx wid recipients = do
   where
     tl = ctx ^. transactionLayer @t @k
 
+
+-- | Estimate a wallet migration cost by estimating fee by doing migration
+-- coin selection first
+computeWalletMigrationCostCoinSels
+    :: forall ctx s t k .
+        ( HasTransactionLayer t k ctx
+        , HasDBLayer s t k ctx
+        , DefineTx t
+        )
+    => ctx
+    -> WalletId
+    -> ExceptT ErrNoSuchWallet IO (Word64,[CoinSelection])
+computeWalletMigrationCostCoinSels ctx wid = do
+    (wal, _, pending) <- readWallet @ctx @s @t @k ctx wid
+    let bp = blockchainParameters wal
+    let utxo = availableUTxO @s @t pending wal
+    let cOpts = coinSelOpts tl (bp ^. #getTxMaxSize)
+    let fOpts = feeOpts tl (bp ^. #getFeePolicy)
+    let batchSize = fromIntegral $ idealBatchSize cOpts
+    let coinSels = selectCoinsForMigration fOpts batchSize utxo
+    let fees = fmap (estimateFee fOpts) coinSels
```

Comment:
Although this computation should lead to the same result, I'd rather compute fee as the sum of `inputBalance sel - changeBalance sel` for all coin selections so we really reflect the actual fee!

Reply:
corrected